### ISSUE TITLE
Usage with themes

### ISSUE THREAD

**Issue by jimthedev** [Microsoft/typescript-styled-plugin#21]
Hi there, great plugin. Only issue I've run into is that I have to jump through a few hoops to get it to work with styled components themes since they require some type composition using the & operator. This means wrapping the ES6 string template in a function which seems to break the plugin's syntax highlighting.

Take the following example where Spinner gets syntax highlighting properly but Spinner2 does not:

```typescript
import themed, { withProps, keyframes } from './themed-components';

interface IProps {
  active?: boolean;
}

// Any html styled components can go here
// Req'd due to: 
const styled = {
  div: withProps<IProps>()(themed.div)
};

// take a type and a string, return a styled component

export const Rotate360 = keyframes`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`;

export const Spinner = styled.div`
  animation: ${Rotate360} 1.1s infinite linear;
  border-color: ${props => props.theme.color.blue400} ${props =>
  props.theme.color.blue400} ${props => props.theme.color.blue400} ${props =>
  props.theme.color.blue200};
  border-radius: 50%;
  border-style: solid;
  border-width: 1.1em;
  display: inline-block;
  font-size: 10px;
  height: 10em;
  margin: 60px auto;
  overflow: hidden;
  position: relative;
  text-indent: -9999px;
  transform: translateZ(0);
  width: 10em;
`;

export const Spinner2 = withProps<IProps>()(themed.div)`
animation: ${Rotate360} 1.1s infinite linear;
border-color: ${props => props.theme.color.blue400} ${props =>
props.theme.color.blue400} ${props => props.theme.color.blue400} ${props =>
props.theme.color.blue200};
border-radius: 50%;
border-style: solid;
border-width: 1.1em;
display: inline-block;
font-size: 10px;
height: 10em;
margin: 60px auto;
overflow: hidden;
position: relative;
text-indent: -9999px;
transform: translateZ(0);
width: 10em;
`;

```

themed-components looks like this:

```typescript
import * as styledComponents from 'styled-components';

import { ThemedStyledComponentsModule } from 'styled-components';
import { ThemedStyledFunction } from 'styled-components';

import { ITheme } from './theme/standard';

function withProps<U>() {
    return <P,T,O>(
        fn: ThemedStyledFunction<P, T, O>
    ): ThemedStyledFunction<P & U, T, O & U> => fn
}

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
  withTheme,
} = styledComponents as ThemedStyledComponentsModule<ITheme>;

export {
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
  withProps,
  ITheme,
  withTheme,
};
export default styled;

```

Effectively I am just wrapping styled-components in themed-components so that I can provide a theme using ThemeProvider.  It actually works really well but there is one caveat, the vs code plugin requires that I have this broken out in my file because it seems not to like composition around the styled template tag:

```
const styled = {
  div: withProps<IProps>()(themed.div)
};
```

If I try to inline those lines in the string template like in Spinner2 then css formatting is gone. It seems like it really wants styled.div syntax right by the template syntax.

**Comment by mjbvz** [comment]
Yes it looks like even after configuring `withProps` as a valid tag, we do not show intellisense in the string:

```
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES6",
        "plugins": [
            {"name": "typescript-styled-plugin", "tags": ["styled", "withProps"]}
        ]
    },
    "exclude": [
        "node_modules",
        "**/node_modules/*"
    ]
}
```

![screen shot 2017-10-24 at 4 38 20 pm](https://user-images.githubusercontent.com/12821956/31973232-c461ae96-b8d9-11e7-8cb3-380cf2e87f34.png)

**Comment by aczekajski** [comment]
@jimthedev - besides the plugin being unable to take "withProps" as proper opening tag, are you sure you need a function like `withProps`? Recently I noticed that styled components (be it `styled.div` or `styled(Something)`) can properly take underlying component props (div props or `Something`'s props) together with its own props if you properly provide a type for a props-taking function in styles. If you write
```ts
interface SthProps {
  whatever: boolean;
}
const Sth = styled.button`
  ${(props: SthProps) => props.whatever ? 'color: red;' : 'background: red;'}
`;
```
then you can properly get props type checking and use it like this:
```html
<Sth whatever={true} onChange={/* ... */} />
```

**Comment by mjbvz** [comment]
Fixed as part of #56 I believe. You still have to configure `tags` but intellisense should now work after that
